src/main/java/github/paz/awardportal/model/User/User.java

```diff
@@ -85,4 +89,51 @@ public void updateUser(BaseUser update){
 
     public User() {
     }
+
+
+    @Override
+    public String toString() {
+        return "User{" +
+                "id=" + id +
+                ", firstName='" + firstName + '\'' +
+                ", lastName='" + lastName + '\'' +
+                ", email='" + email + '\'' +
+                ", password='" + password + '\'' +
+                ", office=" + office +
+                ", isAdmin=" + isAdmin +
+                ", signature=" + Arrays.toString(signature) +
+                ", receivedAwards=" + receivedAwards +
+                '}';
+    }
+
+    // set signature from Multipartfile format which supports upload using <form>
+    public void setSignature(MultipartFile file) {
+        try {
+            this.signature = file.getBytes();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    // return the signature in base64 string, allow it to be displayed in html
+    public String getEncodedSignature() {
```

Comment:
Looks good. However, what do you think about this: https://www.baeldung.com/spring-boot-jsoncomponent

We can customize how User objects are serialized to JSON such that the signature field, if present, is automatically converted to a Base64 string. That way, the front end just needs to access this field <img src="data:image/jpg;base64, {{user.signature}} />.

Reply:
Looks like JSON outputs encoded string by default in `encodedSignature` field.